ika/client.go

```diff
@@ -78,3 +85,86 @@ func (p *Client) UpdateLC(
 		RequestType: "WaitForLocalExecution",
 	})
 }
+
+// ApproveAndSign approves and signs a set of messages using the IKA network. Returns its signatures
+func (p *Client) ApproveAndSign(
+	ctx context.Context,
+	dwalletCapID string,
+	signMessagesID string,
+	messages [][]byte,
+) ([][]byte, error) {
+
+	// TODO: This function was only tested against dummy implementation of the dwallet module deployed locally.
+	// Once it is ready, test it again
+	req := models.MoveCallRequest{
+		Signer:          p.Signer.Address,
+		PackageObjectId: p.LcPackage,
+		Module:          p.Module,
+		Function:        "approve_messages",
+		TypeArguments:   []interface{}{},
+		Arguments: []interface{}{
+			dwalletCapID,
+			messages,
+		},
+		GasBudget: p.GasBudget,
+	}
+	messageApprovals, err := p.c.MoveCall(ctx, req)
+	if err != nil {
+		return nil, fmt.Errorf("error calling approve_messages function: %w", err)
+	}
+
+	req.Function = "sign"
+	req.TypeArguments = []interface{}{}
+	req.Arguments = []interface{}{
+		signMessagesID,
+		messageApprovals,
+	}
+	resp, err := p.c.MoveCall(ctx, req)
+	if err != nil {
+		return nil, fmt.Errorf("error calling sign function: %w", err)
+	}
+
+	response, err := p.c.SignAndExecuteTransactionBlock(ctx, models.SignAndExecuteTransactionBlockRequest{
+		TxnMetaData: resp,
+		PriKey:      p.Signer.PriKey,
+		Options: models.SuiTransactionBlockOptions{
+			ShowInput:    true,
+			ShowRawInput: true,
+			ShowEffects:  true,
+		},
+		RequestType: "WaitForLocalExecution",
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error executing transaction block: %w", err)
+	}
+
```

Comment:
I see we are using this part of logic a few times. Let's refactor it. 
Notes: I suggest you do this in other PRs, and this is not a high priority.